### PR TITLE
BUG: item_cache invalidation in get_numeric_data

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/methods/test_cov_corr.py [hunk 1]
```diff
@@ -191,6 +191,23 @@ def test_corr_nullable_integer(self, nullable_column, other_column, method):
         expected = pd.DataFrame(np.ones((2, 2)), columns=["a", "b"], index=["a", "b"])
         tm.assert_frame_equal(result, expected)
 
+    def test_corr_item_cache(self):
+        # Check that corr does not lead to incorrect entries in item_cache
```
Comment (WillAyd) [inline pandas/tests/frame/methods/test_cov_corr.py:195]: Is there an open issue for this? If not can you add a whatsnew?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/methods/test_cov_corr.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_cov_corr.py:195]: whatsnew added

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -719,7 +719,6 @@ def get_numeric_data(self, copy: bool = False) -> "BlockManager":
         copy : bool, default False
             Whether to copy the blocks
         """
-        self._consolidate_inplace()
```
Comment (jreback) [inline pandas/core/internals/managers.py:722]: if you are removing this then get_bool_data right above looks suspicious.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:722]: i agree.  thats part of the _reduce trouble we discussed this afternoon.  #34918 removes that usage.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by simonjayhawkins** [comment]
whatsnew in 1.1.2. should this be backported and milestone changed or release note moved?

in release readiness I now check whatsnew on master and backport in sync

https://github.com/simonjayhawkins/pandas-release/runs/1078142445?check_suite_focus=true

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**